### PR TITLE
Sara Chandler -- Carets

### PR DESCRIPTION
**Pull request by SaraChandler**
# API Muncher
Congratulations! You're submitting your assignment!

## Comprehension Questions
| Question | Answer    |
| :------------- | :------------- |
| How did you go about exploring the Edamam API, how did you try querying the API? |I read the API docs and tried searching with postman.  |
| Describe your API Wrapper.  How did you decide on the methods you created? | My wrapper has two methods. I knew we would need to return a list of recipes for the index page, so I made a method that would return an array of recipe objects. My second method takes the URI data from a recipe object and returns the recipe so it can be viewed in the show page. |
| Describe an edge case or failure case test you wrote for your API Wrapper. | If the search returns no results, the wrapper will return an empty array. |
| Explain how VCR aids in testing an API. | VCR is used to reduce dependency on APIs when testing. With it we can record HTTP interactions and replay them when needed, so we don't have to make those calls to the API every time we want to run a test. |
| What is the Heroku URL of your deployed application? | https://guarded-waters-62053.herokuapp.com/ |

### CODE REVIEW

File: app/views/recipes/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+</div>
+
+<section class="results row large-12">
+  <h2 class="indexhead"><%= params[:search].capitalize%> Recipe List</h2>
+
+<ul class="row recipe-list">
+    <% @results_page.each do |recipe| %>
+    <article class="results_list column float-left large-4 medium-6 small-12">
+    <li>
+        <%= image_tag recipe.image, alt: "foods" %>
```
Comment (CheezItMan) [inline app/views/recipes/index.html.erb:17]: It would make sense for this image to be a link to the recipe too!

File: lib/edamam_api_wrapper.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        recipes_hash["recipe"]["dietLables"])
+      end
+
+      return my_recipes
+    else
+      return []
+    end
+  end
+
+  def self.show_recipe(uri)
```
Comment (CheezItMan) [inline lib/edamam_api_wrapper.rb:28]: Some error detection here should be done, what if response[0] is `nil`?

File: test/lib/edamam_api_wrapper_test.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      VCR.use_cassette("get_recipes") do
+        search = ""
+        recipes = EdamamApiWrapper.search_recipes(search)
+
+        recipes.must_equal []
+      end
+    end
+  end
+
+  describe "show_recipe" do
```
Comment (CheezItMan) [inline test/lib/edamam_api_wrapper_test.rb:30]: What about negative case tests?  What if the item put in the URL is wrong?

File: test/controllers/recipes_controller_test.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    it "can show a recipe that exists" do
+      VCR.use_cassette("recipes") do
+        recipe_uri = "http://www.edamam.com/ontologies/edamam.owl%23recipe_f1c853a77986214680bbdd424883499a"
+        get recipe_show_path("Herbes de Provence Rotisserie Chickens"), params:{uri: recipe_uri}
+
+        must_respond_with :success
+      end
+    end
+
+    # it "will render 404 if recipe isn't found" do
```
Comment (CheezItMan) [inline test/controllers/recipes_controller_test.rb:42]: Good that you thought of it, this test exposes a problem with your wrapper.

**Comment by CheezItMan** [comment]
# API Muncher
## What We're Looking For

| Feature | Feedback    |
| :------------- | :------------- |
| **Core Requirements** |   |
| Git hygiene | Good commit messages, but more granular commits would be better.  |
| Comprehension questions	| Check |
| **General** |  |
| Rails fundamentals (RESTful routing, use of named paths) | Check |
| Semantic HTML | Check |
| Errors are reported to the user | Check for no results on a search, but if I give it an invalid recipe URI the page crashes. |
| API Wrapper to handle the API requests | Check |
| Controller testing | Yes, but you commented out an important negative-case test.  See my in-code comments. |
| Lib testing | Check, but again no negative-case testing. |
| Search Functionality | Check |
| List Functionality | Check |
| Show individual item functionality (link to original recipe opens in new tab) | The link is there, but it doesn't open in a new tab. |
| **Styling** |  |
| Foundation Styling for responsive layout | Check, well done |
| List View shows 10 items at a time/pagination | Chekc |
| The app is styled to create an attractive user interface | the interface is ok, but the cat image pushes the search bar very far down and the show layout is pretty basic. |
| **API Features** | |
| The App attributes Edaman | The attribution is there, but not linking back to Edamam |
| The VCR casettes do not contain the API key | Check |
| **External Resources** |  |
| Link to deployed app on Heroku | Check, well done  |
|  **Overall** | You hit the requirements, but missed some negative-case testing and your styling was rather basic.  Overall, not bad. |
